pyhive/tests/sqlalchemy_test_case.py

```diff
@@ -53,62 +56,96 @@ def test_reflect_no_such_table(self, engine, connection):
-        engine.dialect.reflecttable(
-            connection, one_row_complex, include_columns=['int'],
-            exclude_columns=[], resolve_fks=True)
+
+        one_row_complex = Table('one_row_complex', MetaData())
+
+        if sqlalchemy_version == 1.3:
+            engine.dialect.reflecttable(
+                connection, one_row_complex, include_columns=['int'],
+                exclude_columns=[], resolve_fks=True)
+        else:
+            insp = sqlalchemy.inspect(engine)
+            insp.reflect_table(
+                one_row_complex,
+                include_columns=['int'],
+                exclude_columns=[],
+                resolve_fks=True,
+            )
+
         self.assertEqual(len(one_row_complex.c), 1)
         self.assertIsNotNone(one_row_complex.c.int)
         self.assertRaises(AttributeError, lambda: one_row_complex.c.tinyint)
 
     @with_engine_connection
     def test_reflect_with_schema(self, engine, connection):
-        dummy = Table('dummy_table', MetaData(bind=engine), schema='pyhive_test_database',
-                      autoload=True)
+        dummy = Table('dummy_table', MetaData(schema='pyhive_test_database'), autoload_with=engine)
         self.assertEqual(len(dummy.c), 1)
         self.assertIsNotNone(dummy.c.a)
 
     @pytest.mark.filterwarnings('default:Omitting:sqlalchemy.exc.SAWarning')
     @with_engine_connection
     def test_reflect_partitions(self, engine, connection):
         """reflecttable should get the partition column as an index"""
-        many_rows = Table('many_rows', MetaData(bind=engine), autoload=True)
+        many_rows = Table('many_rows', MetaData(), autoload_with=engine)
         self.assertEqual(len(many_rows.c), 2)
         self.assertEqual(repr(many_rows.indexes), repr({Index('partition', many_rows.c.b)}))
 
-        many_rows = Table('many_rows', MetaData(bind=engine))
-        engine.dialect.reflecttable(
-            connection, many_rows, include_columns=['a'],
-            exclude_columns=[], resolve_fks=True)
+        many_rows = Table('many_rows', MetaData())
+
+        if sqlalchemy_version == 1.3:
+            engine.dialect.reflecttable(
+                connection, many_rows, include_columns=['a'],
+                exclude_columns=[], resolve_fks=True)
+        else:
+            insp = sqlalchemy.inspect(engine)
+            insp.reflect_table(
+                many_rows,
+                include_columns=['a'],
+                exclude_columns=[],
+                resolve_fks=True,
+            )
+        
         self.assertEqual(len(many_rows.c), 1)
         self.assertFalse(many_rows.c.a.index)
         self.assertFalse(many_rows.indexes)
 
-        many_rows = Table('many_rows', MetaData(bind=engine))
-        engine.dialect.reflecttable(
-            connection, many_rows, include_columns=['b'],
-            exclude_columns=[], resolve_fks=True)
+        many_rows = Table('many_rows', MetaData())
+
+        if sqlalchemy_version == 1.3:
```

Comment:
let's add a utility function to hide code duplication

Reply:
done